library/public/interactor.h

```diff
@@ -23,21 +24,36 @@ class F3D_EXPORT interactor
 {
 public:
   ///@{ @name Command
+  /**
+   * Remove all existing command callbacks and create all default command callbacks,
+   * see COMMANDS.md for details.
+   */
+  virtual interactor& createDefaultCommandCallbacks() = 0;
+
   /**
    * Use this method to add a callback into the command map
    * to be called using triggerCommand.
-   * Adding a commandCallback with an existing action replaces it.
+   * Adding a commandCallback with an already existing action throw a
+   * interactor::already_exists_exception.
+   * Considering namespacing dedicated action to avoid conflicts with default action,
+   * eg: `my_app::action`
    */
   virtual interactor& addCommandCallback(
-    std::string action, std::function<bool(const std::vector<std::string>&)> callback) = 0;
+    const std::string& action, std::function<bool(const std::vector<std::string>&)> callback) = 0;
 
   /**
-   * Remove a command callback for provided action
+   * Remove a command callback for provided action, does not do anything if it does not exists.
    */
   virtual interactor& removeCommandCallback(const std::string& action) = 0;
 
   /**
-   * Trigger provided command, see COMMAND.md for more information
+   * Return a string vector containing all currently defined actions of command callbacks
+   */
+  virtual std::vector<std::string> getCommandCallbackActions() = 0;
```

Comment:
not sure about naming here, it could be 

`getCommandActions` instead.

Reply:
Maybe just `addCommand`/`removeCommand`/`getCommands`?  
The term `callback` is not really needed? Sounds obvious to me.

Reply:
then `addInteractionCommand` is ambiguous imo

Reply:
if `addCommandCallback` can be `addCommand` then maybe `addInteractionCommand` could be `addInteraction` according to the same logic of the value type being obvious/implicit?

Reply:
im not against it, wdyt @Meakk ?

Reply:
I'd say `addBinding` instead of `addInteraction`. `interactor.addInteraction` sounds bad to me.